cmd/dev/app/rule_type/init.go

```diff
@@ -0,0 +1,188 @@
+// SPDX-FileCopyrightText: Copyright 2023 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package rule_type
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/spf13/cobra"
+)
+
+// CmdInit is the command for initializing a rule type definition
+func CmdInit() *cobra.Command {
+	initCmd := &cobra.Command{
+		Use:   "init",
+		Short: "initialize a rule type definition",
+		Long: `The 'ruletype init' subcommand allows you to initialize a rule type definition
+
+The first positional argument is the directory to initialize the rule type in.
+The rule type will be initialized in the current directory if no directory is provided.
+`,
+		RunE:         initCmdRun,
+		SilenceUsage: true,
+	}
+
+	initCmd.Flags().StringP("name", "n", "", "name of the rule type")
+	initCmd.Flags().BoolP("skip-tests", "s", false, "skip creating test files")
+
+	if err := initCmd.MarkFlagRequired("name"); err != nil {
+		fmt.Fprintf(os.Stderr, "Error marking flag as required: %s\n", err)
+		os.Exit(1)
+	}
+
+	return initCmd
+}
+
+func initCmdRun(cmd *cobra.Command, args []string) error {
```

Comment:
What do you think about having this operate in a more "wizard" style in terms of prompting for missing data rather than people needing to figure out the flag arguments up front and then getting a file with a bunch of comments to fill in?